version: '3.8'

services:
    postgres:
        image: postgres:12.4-alpine
        restart: on-failure
        environment:
            POSTGRES_DB: playme
            POSTGRES_USER: playme_usr
            POSTGRES_PASSWORD: 2345playMePasS35
            PGDATA: /var/lib/postgresql/data
        volumes:
            - db-data:/var/lib/postgresql/data:rw
        ports:
            - "5432:5432"

    php:
        build:
            context: .
            target: app_php #target the app_php section in ./Dockerfile
        image: sf-docker/php:dev #path-to-custom-image:tag
        restart: on-failure
        environment:
            APP_ENV: dev
            APP_DEBUG: 1
            PHP_DATE_TIMEZONE: ${PHP_DATE_TIMEZONE:-UTC}
            XDEBUG_CONFIG: remote_host=docker.for.mac.localhost
            PHP_IDE_CONFIG: serverName=localhost
        depends_on:
            - postgres
        volumes:
            - .:/app:rw,cached
            - var:/app/var

    nginx:
        build:
            context: .
            target: app_nginx #target the app_nginx section in ./Dockerfile
        image: sf-docker/nginx:dev #path-to-custom-image:tag
        restart: on-failure
        depends_on:
            - php
        volumes:
            - ./public:/app/public:ro
        ports:
            - "8080:80"

volumes:
    db-data:
    var: